packages/fether-electron/src/main/app/index.js

```diff
@@ -0,0 +1,108 @@
+// Copyright 2015-2018 Parity Technologies (UK) Ltd.
+// This file is part of Parity.
+//
+// SPDX-License-Identifier: BSD-3-Clause
+
+import events from 'events';
+
+import {
+  setupAppListeners,
+  createWindow,
+  updateProgress,
+  createPositioner,
+  setupRequestListeners,
+  createTray,
+  loadTray,
+  showTrayBalloon,
+  setupDebug,
+  setupSecurity,
+  setupLogger,
+  setupParityEthereum,
+  setupGlobals,
+  setupMenu,
+  getScreenResolution,
+  calculateWindowPosition,
+  onTrayClick,
+  fixWindowPosition,
+  showWindow,
+  moveWindowUp,
+  processSaveWindowPosition,
+  hideWindow,
+  windowClear,
+  onWindowClose,
+  setupWindowListeners,
+  setupWin32Listeners
+} from './methods';
+
+let hasCalledInitFetherApp = false;
+
+class FetherApp {
+  fetherApp = new events.EventEmitter();
+
+  constructor (electronApp, options) {
+    if (hasCalledInitFetherApp) {
+      this.fetherApp.emit(
+        'error',
+        new Error('Unable to initialise Fether app more than once')
+      );
+    }
+
+    this.fetherApp.app = electronApp;
+    this.fetherApp.options = options;
+  }
+}
+
+FetherApp.prototype.setupAppListeners = setupAppListeners;
```

Comment:
It's 2019, let's avoid using prototypes.

I'd recommand, in the constructor to do:
```
setupAppListeners(this);
// ... same for all other methods, i.e pass the instance as an argument to the function
```

Reply:
I tried implementing `class FetherApp extends EventEmitter`, adding `super()` in the constructor, and renaming occurrences of `this.fetherApp.emit(...` with `this.emit(...`

I also tried renaming `events` to `EventEmitter`, removing `fetherApp = new events.EventEmitter();` and replacing it with `this.fetherApp = new EventEmitter.EventEmitter();` after `super()`,
but although it didn't generate an error, it didn't appear to ever compile.

Reply:
> ```
> setupAppListeners(this); // or setupAppListeners(this.fetherApp), not sure
> // ... same for all other methods, i.e pass the instance as an argument to the function
> ```

I'll have to pass `this` as an argument, so the methods can call other methods can call them

Reply:
Update: I got the event emitter refactoring to work after I focused on fixing everything else first